bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/ApiBridge.java

```diff
@@ -0,0 +1,227 @@
+            connectApi.authenticate();
+            setConnectionStatus(ConnectionStatus.SUCCESS);
+        } catch (NetatmoException e) {
+            prepareReconnection(e);
+        }
+    }
+
+    @Deactivate
+    public void dispose() {
+        connectApi.dispose();
+        freeConnectJob();
+    }
+
+    private void freeConnectJob() {
+        ScheduledFuture<?> job = connectJob;
+        if (job != null) {
+            job.cancel(true);
+        }
+        connectJob = null;
+    }
+
+    private void prepareReconnection(NetatmoException e) {
+        setConnectionStatus(ConnectionStatus.Failed("Will retry to connect Netatmo API, this call failed : %s", e));
+        onAccessTokenResponse(null, List.of());
+        freeConnectJob();
+        connectJob = scheduler.schedule(() -> openConnection(null), configuration.reconnectInterval, TimeUnit.SECONDS);
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T extends RestManager> T getRestManager(Class<T> typeOfRest) {
+        if (!managers.containsKey(typeOfRest)) {
+            try {
+                Constructor<T> constructor = typeOfRest.getConstructor(ApiBridge.class);
+                T tentative = constructor.newInstance(this);
+                if (!grantedScopes.containsAll(tentative.getRequiredScopes())) {
+                    throw new NetatmoException("Required scopes missing to access : " + typeOfRest);
+                }
+                managers.put(typeOfRest, tentative);
+            } catch (SecurityException | ReflectiveOperationException | NetatmoException e) {
+                logger.error("Error invoking RestManager constructor for class {} : {}", typeOfRest, e.getMessage());
+            }
+        }
+        return (T) managers.get(typeOfRest);
+    }
+
+    synchronized <T> T executeUri(URI uri, HttpMethod method, Class<T> classOfT, @Nullable String payload)
+            throws NetatmoException {
+        try {
+            logger.debug("executeUri {}  {} ", method.toString(), uri);
+
+            Request request = httpClient.newRequest(uri).method(method).timeout(TIMEOUT_MS, TimeUnit.MILLISECONDS);
+
+            httpHeaders.entrySet().forEach(entry -> request.header(entry.getKey(), entry.getValue()));
+
+            if (payload != null && (HttpMethod.POST.equals(method) || HttpMethod.PUT.equals(method))) {
+                InputStream stream = new ByteArrayInputStream(payload.getBytes(StandardCharsets.UTF_8));
+                try (InputStreamContentProvider inputStreamContentProvider = new InputStreamContentProvider(stream)) {
+                    request.content(inputStreamContentProvider, null);
+                }
+                if (!NetatmoConstants.URL_API.contains(uri.getHost())) {
+                    request.getHeaders().remove(HttpHeader.CONTENT_TYPE);
+                    request.header(HttpHeader.CONTENT_TYPE, "application/json;charset=utf-8");
+                }
+            }
+
+            ContentResponse response = request.send();
```

Comment:
Don't know if useful, but consider if request and response should be logged with log level TRACE.

Reply:
Correct.